docs/connector-development/connector-builder-ui/authentication.md

```diff
@@ -0,0 +1,150 @@
+
+This form of authentication is often called "(custom) header authentication".
+
+#### Example
+
+The [CoinAPI.io API](https://docs.coinapi.io/market-data/rest-api#authorization) is using API key authentication via the `X-CoinAPI-Key` header.
+
+When fetching records, the api token is included in the request using the configured header:
+```
+curl -X GET \
+  -H "X-CoinAPI-Key: <api-key>" \
+  https://rest.coinapi.io/v1/<stream path>
+```
+
+### OAuth
+
+The OAuth authentication method implements authentication using an [OAuth2.0 flow with a refresh token grant type](https://oauth.net/2/grant-types/refresh-token/).
+
+In this scheme the OAuth endpoint of an API is called with a long-lived refresh token provided as part of the source configuration to obtain a short-lived access token that's used to make requests actually extracting records. If the access token expires, the connection will automatically request a new one.
+
+The connector needs to be configured with the endpoint to call to obtain access tokens with the refresh token. OAuth client id/secret and the refresh token are provided via "Testing values" in the connector builder as well as part of the source configuration when configuring connections.
+
+Depending on how the refresh endpoint is implemented exactly, additional configuration might be necessary to specify how to request an access token with the right permissions (configuring OAuth scopes and grant type) and how to extract the access token and the expiry date out of the response (configuring expiry date format and property name as well as the access key property name):
+* Scopes - if not specified, no scopes are sent along with the refresh token request
+* Grant type - if not specified, it's set to `refresh_token`
+* Token expiry property name - if not specified, it's set to `expires_in`
+* Token expire property date format - if not specified, the expiry property is interpreted as the number of seconds the access token will be valid
+* Access token property name - the name of the property in the response that contains the access token to do requests. If not specified, it's set to `access_token`
+
+If the API uses a short-lived refresh token that expires after a short amount of time and needs to be refreshed as well or if other grant types like PKCE are required, it's not possible to use the connector builder with OAuth authentication - check out the [compatibility guide](/connector-development/config-based/connector-builder-compatibility#oauth) for more information.
+
+Keep in mind that the OAuth authentication method does not implement a single-click authentication experience for the end user configuring the connector - it will still be necessary to obtain client id, client secret and refresh token from the API and manually enter them into the configuration form.
```

Comment:
Would it make sense here to explain why this isn't supported? E.g. something along the lines of "Single-click authentication is only supported in Airbyte Cloud. For more information on Airbyte Cloud, click here [link]. To get your connector added to the Airbyte Cloud connector catalog, see here [link]"
(not sure if we have those other pages to link to at the moment)

This would help advertise Airbyte Cloud and explain a bit why it isn't in OSS. But if we don't want this to be any sort of advertisement for Airbyte cloud (which may make sense in concept docs like these), then I would be fine with leaving this out too

Reply:
Single click authentication won't be available on airbyte cloud for custom connectors because we need to implement a custom oauth flow for each new connector